### PR TITLE
Update/stamping

### PR DESCRIPTION
**Pull request by JonathanPrince**
adds customizable text to footer stamp

### CODE REVIEW

**Comment by JonathanPrince** [comment]
@digitalsadhu 👀  ?

File: lib/middleware.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  setupTemp (req, res, next) {
+    debug('setup temp storage')
+    req.temp = tmp.dirSync({unsafeCleanup: true})
+    next()
+  },
+
+  writeHtmlFile (req, res, next) {
+    debug('write html file')
+    fs.writeFile(`${req.temp.name}/html.html`, req.body.html, err => {
+      if (err) res.status(500).send(err)
```
Comment (digitalsadhu) [inline lib/middleware.js:33]: you probably shouldn't be handling errors in your middleware. You should call next(err) and let error handling middleware do the job of catching all errors, generating html user friendly errors

Comment (digitalsadhu) [inline lib/middleware.js:33]: ps. you should add error handling middleware.

File: lib/middleware.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      const args = [
+        '-dBATCH',
+        '-dNOPAUSE',
+        '-q',
+        '-sDEVICE=pdfwrite',
+        `-sOutputFile=${req.temp.name}/output.pdf`,
+        `${req.temp.name}/stamp.ps`,
+        `${req.temp.name}/pre-stamp.pdf`
+      ]
+      execa('gs', args)
```
Comment (digitalsadhu) [inline lib/middleware.js:146]: happy with execa?

File: index.js [hunk 3]
```diff
@@ -1,93 +1,23 @@
...
-  })
-})
+app.use(mw.setupTemp)
+app.use(mw.writeHtmlFile)
+app.use(mw.convertHtmlToPdf)
+app.use(mw.createFileList)
+app.use(mw.mergeFileList)
+app.use(mw.createPostscript)
+app.use(mw.addStamp)
+
```
Comment (digitalsadhu) [inline index.js:20]: error middleware here, should be registered last after other middleware

**Comment by digitalsadhu** [comment]
Really liking the structure.

**Comment by JonathanPrince** [comment]
thanks for the feedback @digitalsadhu, will address the error handling in a new PR.

**Pull request merged**